src/qibolab/_core/instruments/emulator/emulator.py

```diff
@@ -89,32 +89,45 @@ def _single_sequence(
         result for the execution of this single sequence, thus suitable
         to be returned as is.
         """
-        # probabilities for the |0>, |1> ... |n> state, for each swept value
+        # probabilities for states in computational basis
         probabilities = self._sweep(sequence, configs, sweepers)
         assert options.nshots is not None
-        # extract results from probabilities, according to the requested averaging mode
-        averaged = (
-            shots(probabilities, options.nshots)
-            if options.averaging_mode == AveragingMode.SINGLESHOT
-            # weighted averaged
-            else np.sum(probabilities * np.arange(probabilities.shape[-1]), axis=-1)
-        )
-        # move measurements dimension to the front, getting ready for extraction
-        res = np.moveaxis(averaged, -1, 0)
-
-        if options.acquisition_type is AcquisitionType.DISCRIMINATION:
-            measurements = res
-        elif options.acquisition_type is AcquisitionType.INTEGRATION:
-            x = np.stack((res, np.zeros_like(res)), axis=-1)
-            measurements = np.random.normal(x, scale=0.001)
-        else:
-            raise ValueError(
-                f"Acquisition type '{options.acquisition_type}' unsupported"
+        results = {}
+        levels = np.array(configs["hamiltonian"].transmon_levels)
+        for ro_id in self._acquisitions(sequence):
+            # TODO: find better to extract qubit from acquisition channel
+            # possible solution _single_sequence should be aware of platform.qubits
+            qubit = int(sequence.pulse_channels(ro_id)[0][0])
+            single_qubit_probabilities = np.moveaxis(
+                np.moveaxis(probabilities, -1, 0)[
+                    qubit * levels : (qubit + 1) * levels
+                ],
+                0,
+                -1,
+            )
```

Comment:
With or without the whole loop, here you're using `np.moveaxis` just to index the last dimension as if it were the first one, and move it back to its place right after.

However, NumPy is very expressive and concise about indexing. If you want to omit a single dimension, just use `:`. If you want to omit as many as you need to get to the last one, just use an ellipsis
```suggestion
            single_qubit_probabilities = probabilities[
                ..., qubit * levels : (qubit + 1) * levels
            ]
```